modules/data.atmosphere/R/met.process.R

```diff
@@ -113,7 +177,34 @@ met.process <- function(site, input_met, start_date, end_date, model, host, dbpa
 
         cmdFcn  = paste0(pkg,"::",fcn,"(",paste0("'",args,"'",collapse=","),")")
         new.files <- remote.execute.R(cmdFcn,host$name,user=NA, verbose=TRUE)
-
+        }
+      }
+      if(input.info$stage$update){
+        start_date <- input.info$dates$update$start_date
+        end_date <- input.info$dates$update$end_date
+        ## Change Start and end date of Input record to include entire range of existing dates
+        db.query(paste("UPDATE inputs SET (start_date, end_date) = (",
+                       "'",as.POSIXlt(start_date),"'",
+                       ",",
+                       "'",as.POSIXlt(end_date), "'",
+                       ") WHERE id = "raw.id),con)
+        
+        #loop over individual files and insert them into database
+        raw.id <- dbfile.input.insert(in.path=dirname(new.files$file[1]),
+                                      in.prefix=new.files$dbfile.name[1],
+                                      siteid = site$id,
+                                      startdate = start_date,
+                                      enddate = end_date,
+                                      mimetype=new.files$mimetype[1],
+                                      formatname=new.files$formatname[1],
+                                      parentid = input.id,
+                                      con = con,
+                                      hostname = host$name)
+        
+        if(met %in% "CRUNCEP"){ready.id = raw.id}
+        if(met %in% "GFDL"){ready.id = raw.id}
```

Comment:
These shouldn't be two separate IF's. This case just needs to be added to the met.registry files as a flag (e.g. direct_extract = TRUE for these and false for NARR) so that we don't have to keep adding special cases